AIPscan/Aggregator/database_helpers.py

```diff
@@ -402,3 +405,45 @@ def process_aip_data(aip, mets):
     preservation_files = [file_ for file_ in all_files if file_.use == "preservation"]
     for file_ in preservation_files:
         create_file_object(FileType.preservation, file_, aip.id)
+
+
+def create_fetch_job(datetime_obj_start, timestamp_str, storage_server_id):
+    fetch_job = FetchJob(
+        total_packages=None,
+        total_deleted_aips=None,
+        total_aips=None,
+        download_start=datetime_obj_start,
+        download_end=None,
+        download_directory=f"AIPscan/Aggregator/downloads/{timestamp_str}/",
+        storage_service_id=storage_server_id,
+    )
+    db.session.add(fetch_job)
+    db.session.commit()
+
+    return fetch_job
+
+
+def update_fetch_job(fetch_job_id, processed_packages, total_packages_count):
+    # Count different types of packages
+    total_aips = len([package for package in processed_packages if package.is_aip()])
```

Comment:
Doing this seems a little inefficient, you could instead have a single for loop it seems like and add to a count variable for each sip, dip and aip in processedpackage instead of making an array for each sip dip and aip and then checking the type.

total_sips
total_dips
total_aips
for package in processed_packages
    if package.is_aip() 
        total_aips++
    if package.is_dip()
        total_dips++
        ...

Reply:
True... and it's harder to read too.

Reply:
I don't see the recommended change here. Should it not be implemented?